.github/workflows/rspec_security.yml

```
@@ -0,0 +1,104 @@
+name: Rails RSpec and security
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:12
+        ports:
+          - 5432:5432
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+        # ヘルスチェック項目
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+      chrome:
+        image: selenium/standalone-chrome
+        ports:
+          - "4444:4444"
+        volumes:
+          - /dev/shm:/dev/shm
+        env:
+          BUNDLE_JOBS: "3"
+          BUNDLE_RETRY: "3"
+          PGHOST: db
+
+    steps:
+      # ref:https://github.com/actions/checkout
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+        # ref:https://github.com/actions/setup-ruby
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.6
+
+      # ref:https://github.com/actions/cache/blob/main/examples.md#ruby---bundler
+      - name: Ruby Bundler
+        uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-
+
+      - name: Bundle install
+        run: |
+          bundle config path vendor/bundle
+          bundle install --jobs 4 --retry 3
+
+        # yarnキャッシュのディレクトリ指定
+        # https://github.com/actions/cache/blob/main/examples.md#node---yarn
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+        # yarnのインストール
+      - name: yarn install
+        run: yarn install
+
+        # yarnのキャッシュ
+        # ref:https://github.com/actions/cache/blob/main/examples.md#node---yarn
+      - uses: actions/cache@v2
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      # DBのセット
+      # db:schemaを使用
+      - name: Setup test database
+        env:
+          RAILS_ENV: test
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+        run: |
+          cp config/database.yml.github-actions config/database.yml
+          bundle exec rake db:create db:schema:load --trace
+
+      # rspecによるテスト
+      # upload-artifactを使用してfailure時にデータを保存
+      - name: run rspec
+        run: bundle exec rspec
```

勘だけど、ここじゃなかろうか。(そもそもRAILS_ENV: testをこのstepで設定しなくても動くのがよく分からない。各jobで設定した環境変数は共通化される？)

```suggestion
        env:
          RAILS_ENV: test
          SELENIUM_REMOTE_URL: http://localhost:4444/wd/hub
        run: bundle exec rspec
```

---

単純にRAILS_ENVが設定されてない場合は、test入れるようにしてるからか。
https://github.com/Tomoaki-Takeuchi-bot/Ehon-to/blob/4d867653b7d733186c09cf506905c8061919478e/spec/rails_helper.rb#L3